sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/hash.scala

```diff
@@ -389,9 +389,15 @@ abstract class HashExpression[E] extends Expression {
       input: String,
       result: String,
       fields: Array[StructField]): String = {
-    fields.zipWithIndex.map { case (field, index) =>
+    val hashes = fields.zipWithIndex.map { case (field, index) =>
       nullSafeElementHash(input, index.toString, field.nullable, field.dataType, result, ctx)
-    }.mkString("\n")
+    }
+    val args = if (ctx.INPUT_ROW != null) {
+      Seq(("InternalRow", input), ("InternalRow", ctx.INPUT_ROW))
```

Comment:
sorry, I cannot understand why you need to pass `ctx.INPUT_ROW` as an argument, might you please explain me? Thanks.

Reply:
Good question. I conservatively pass `ctx.INPUT_ROW`. When I revisit this question, I believe that elements in `struct` would not use `ctx.INPUT_ROW`.